src/serialization/form/FormParseNode.cs

```diff
@@ -143,21 +144,21 @@ public IEnumerable<T> GetCollectionOfPrimitiveValues<T>()
         }
     }
     /// <inheritdoc/>
-    public DateTimeOffset? GetDateTimeOffsetValue() => DateTimeOffset.TryParse(DecodedValue, out var result) ? result : null;
+    public DateTimeOffset? GetDateTimeOffsetValue() => DateTimeOffset.TryParse(DecodedValue, CultureInfo.InvariantCulture, DateTimeStyles.None, out var result) ? result : null;
     /// <inheritdoc/>
-    public Date? GetDateValue() => DateTime.TryParse(DecodedValue, out var result) ? new Date(result) : null;
+    public Date? GetDateValue() => DateTime.TryParse(DecodedValue, CultureInfo.InvariantCulture, DateTimeStyles.None, out var result) ? new Date(result) : null;
     /// <inheritdoc/>
-    public decimal? GetDecimalValue() => decimal.TryParse(DecodedValue, out var result) ? result : null;
+    public decimal? GetDecimalValue() => decimal.TryParse(DecodedValue, NumberStyles.Number, CultureInfo.InvariantCulture, out var result) ? result : null;
```

Comment:
shouldn't the style be none here?

Reply:
With None TryParse returns false

Reply:
In specific scenarios or for all scenarios?

Reply:
`None` will work only for numbers without decimals.